plugins/modules/zos_data_set.py

```diff
@@ -1179,7 +1182,7 @@ def parse_and_validate_args(params):
             # ["batch", "space_type"],
             # ["batch", "space_primary"],
             # ["batch", "space_secondary"],
-            ["batch", "record_format"],
+            # ["batch", "record_format"],
```

Comment:
Does this open up a path for ambiguous behavior where `record_format` is defined both inside of a task in `batch` as well as outside?

```yml
- name: Create multiple data sets.
    zos_data_set:
---> record_format: FBA     <------ambiguous??
        batch:
            - name: SOME.DATA.SET
                state: present
                type: PDS
                record_format: FB
                record_length: 80
            - name: ANOTHER.DATA.SET
                state: present
                type: PDS
                record_format: VB
                record_length: 80
```

What is record format of `some.data.set` and `another.data.set`? Are they both `FBA`? Are they `FB` and `VB` respectively? 

With `["batch", "record_format"]` being mutually exclusive, presumably the above task would have failed. But with the new change I'm seeing where the line is commented out (implying those two items are no longer mutually exclusive), my concern is that with such a playbook task as given above, there will be ambiguous behavior. 

From a convenience perspective, maybe not having to define `record_format` for each task in batch and defining it only at the higher level is advantageous. 
For example, say I want to create `SOME.DATA.SET1`, `SOME.DATA.SET2` and `SOME.DATA.SET3` as `FBA` and `ANOTHER.DATA.SET` as `VB`, perhaps I could define a sort of 'default' record format at the higher level and then if record format is differently defined at the nested level, that would override the higher level default:

```yml
- name: Create multiple data sets.
    zos_data_set:
      record_format: FBA
        batch:
            - name: SOME.DATA.SET1
                state: present
                type: PDS
                record_length: 80
            - name: SOME.DATA.SET2
                state: present
                type: PDS
                record_length: 80
            - name: SOME.DATA.SET3
                state: present
                type: PDS
                record_length: 80
            - name: ANOTHER.DATA.SET
                state: present
                type: PDS
----->         record_format: VB   <------ override higher level 'default'
                record_length: 80
```

If so, this would need to be documented and further validated.

Reply:
Actually, it's backwards: 
If a batch is used, ONLY the record_format for the batch entries are used.
If a batch is NOT used, then only the higher (outside of batch) is used, and is a single-shot.

This is because of 2 contradictory requirements: to add choices means NULL/None is not a valid choice, and to use Batch means the outer record_format isn't used.  So, I have it clear the record_format AFTER the validation, based on whether batch is populated or not.

Also, I had to removed the 'exclusive' setting to handle this paradox (record format cannot be blank but does not exist if batch is filled it.)

Reply:
I never realized this wasn't described well in the documentation.  I'll think on it tonight, but there are 2 sets of essentially duplicate settings:

In non-batch/1-shot mode, only outer settings are used (name, type, space type, etc)
In batch mode, settings are completely contained within on batch suboption. (so one can be fb sized in megs and another can be vb sized in tracks and another can be pdse without initial size).


It made sense to me after mucking in the code in this quarter, but it probably needs to be clarified in the actual doc.

Reply:
You were right, looking at the code plus your explanation helped clarify this. The initial delta is why I raised a flag here, but it looks like this won't introduce a completely _new_ behavior. The same way a `record_format` param is now "valid" syntax-wise outside of a batch operation, we already had some others - `space_primary`, `space_type`). 

So the params outside the `batch` call are accepted by the Ansible parser but ignored by our logic. I see this is the way to get those "choices" defined to the ansible parser. 

```yml
- name: create data set
  ibm.ibm_zos_core.zos_data_set:
    space_primary: 100
    space_type: TRK
    record_format: FBA
    batch:
      - name: "some.data.set.one"
        type: SEQ
        record_format: VB
        state: present
        space_primary: 50
        space_type: TRK
      - name: "some.data.set.two"
        type: SEQ
        state: present
        record_format: VB
        space_primary: 50
        space_type: TRK
```

The above is valid syntax, but `space_primary: 100`, `space_type: TRK`, and `record_format: FBA` are ignored. Rather the values within the individual batch items are honored. 

![image](https://github.com/ansible-collections/ibm_zos_core/assets/10567394/8f08410f-94cd-4f59-bd61-84482f4b5860)


Since this is not completely new behavior, I think it's this change is fine.
Ideally, those options outside of the batch would raise an error, but that's an exploration for another day.